pkg/loki/loki.go

```diff
@@ -481,7 +482,24 @@ func (t *Loki) setupModuleManager() error {
 		}
 	}
 
+	t.deps = deps
 	t.ModuleManager = mm
 
 	return nil
 }
+
+func (t *Loki) isModuleEnabled(m string) bool {
```

Comment:
Can we make this `isModuleActive`? We have another function with the same name on the `Config` type which behaves differently: https://github.com/grafana/loki/blob/main/pkg/loki/loki.go#L196-L198

Reply:
Ideally this would be a recursive check so we could handle `IsActive(buzz) == true` when `target=foo`
```
foo: {bar}
bar: {bazz}
bazz: {buzz}
```

Reply:
Renamed and made it recursive

Reply:
Just double checking: there's no chance of Loki having a cyclic dependency, right? Ex:
```yaml
foo: {bar}
bar: {bazz}
bazz: {bar}
```
Because, if there is, I think it could end in an infinite loop (solution would be to track the visited modules and to not visit already visited nodes).

Reply:
haha, i think this could totally happen.